libraries/opensk/src/env/test/mod.rs

```diff
@@ -27,6 +28,13 @@ use rand::rngs::StdRng;
 use rand::{Rng, SeedableRng};
 use rng256::Rng256;
 
+#[cfg(not(feature = "rust_crypto"))]
+pub mod crypto;
+#[cfg(feature = "rust_crypto")]
+pub mod rust_crypto;
+#[cfg(feature = "rust_crypto")]
+pub use rust_crypto as crypto;
```

Comment:
I see 2 alternatives:
1. Move both software crypto implementations to `src/api/crypto`, keep the feature flag, and name that single type `SoftwareCrypto` (or something along those lines).
2. Move both software crypto implementation to `src/api/crypto` but give each one its own name (e.g. `OpenskCrypto` and `RustCrypto`), then use the feature flag here (i.e. `src/env/test`) to merge those names into `TestCrypto`.

In both cases, the Tock environment will reuse the implementations in `src/api/crypto`, either `SoftwareCrypto` or `OpenskCrypto`.

Reply:
I was wondering about moving the Crypto implementation somewhere TockEnv can reuse it. I think we need to keep `RustCrypto` completely behind a feature flag, or we can't use an older compiler. Or is it smart enough to not load optional dependencies if they are unused?
A bit less clean if the Crypto implementation is not in `env/`, but in `api/`, but the reuse is probably worth it.

Reply:
Moved